samza-api/src/main/java/org/apache/samza/container/SamzaContainerContext.java

```diff
@@ -31,20 +32,38 @@
   public final String id;
   public final Config config;
   public final Collection<TaskName> taskNames;
+  public final MetricsRegistry metricsRegistry;
 
   /**
    * An immutable context object that can passed to tasks to give them information
    * about the container in which they are executing.
    * @param id The id of the container.
    * @param config The job configuration.
    * @param taskNames The set of taskName keys for which this container is responsible.
+   * @param metricsRegistry the {@link MetricsRegistry} for the container metrics
    */
   public SamzaContainerContext(
       String id,
       Config config,
-      Collection<TaskName> taskNames) {
+      Collection<TaskName> taskNames,
+      MetricsRegistry metricsRegistry) {
     this.id = id;
     this.config = config;
     this.taskNames = Collections.unmodifiableCollection(taskNames);
+    this.metricsRegistry = metricsRegistry;
+  }
+
+  /**
+   * See also {@link SamzaContainerContext#SamzaContainerContext(String, Config, Collection, MetricsRegistry)}.
+   * NOTE: Used for testing only.
```

Comment:
Can we make this protected or remove public then?

Reply:
Actually the tests in other packages creates container context so I wasn't able to make it pacakge private. But on a second thought, this constructor is kind of confusing and make stuff ugly. I removed it and add the metric registry to all the tests now.